Sources/MultiSets/CountedSet/CountedSet+SetAlgebra.swift

```diff
@@ -0,0 +1,187 @@
+  -> CountedSet<Element> {
+    var result = self
+    result.formUnion(other)
+    return result
+  }
+
+  /// Returns a new set with the lesser number of elements of both this and the
+  /// given set.
+  /// - Parameter other: A set of the same type as the current set.
+  /// - Returns: A new set with the elements of this set and `other`, preserving
+  /// the lower multiplicity.
+  /// - Complexity: O(*k*), where *k* is the number of unique elements in the
+  /// current set.
+  @inlinable
+  public __consuming func intersection(_ other: CountedSet<Element>)
+  -> CountedSet<Element> {
+    var result = self
+    result.formIntersection(other)
+    return result
+  }
+
+
+  /// Returns a new set with the difference between the number of elements of
+  /// both this and the given set.
+  /// - Parameter other: A set of the same type as the current set.
+  /// - Returns: A new set with the elements of this set and `other`, taking the
+  /// difference between multiplicities.
+  @inlinable
+  public __consuming func symmetricDifference(
+    _ other: __owned CountedSet<Element>
+  ) -> CountedSet<Element> {
+    var result = self
+    result.formSymmetricDifference(other)
+    return result
+  }
+
+  /// Inserts the given element in the set.
+  ///
+  /// If an element equal to `newMember` is already contained in the set, its
+  /// multiplicity is incremented by one. In this example, a new element is
+  /// inserted into `classDays`, a set of days of the week.
+  ///
+  ///     enum DayOfTheWeek: Int {
+  ///         case sunday, monday, tuesday, wednesday, thursday,
+  ///             friday, saturday
+  ///     }
+  ///
+  ///     var classDays: CountedSet<DayOfTheWeek> = [.wednesday, .friday]
+  ///     print(classDays.insert(.monday))
+  ///     // Prints "(true, .monday)"
+  ///     print(classDays)
+  ///     // Prints "[.friday, .wednesday, .monday]"
+  ///
+  ///     print(classDays.insert(.friday))
+  ///     // Prints "(true, .friday)"
+  ///     print(classDays)
+  ///     // Prints "[.friday, .friday, .wednesday, .monday]"
+  ///
+  /// - Parameter newMember: An element to insert into the set.
+  /// - Complexity: Amortized O(1)
+  /// - Note: Insertion is always performed, in contrast to an uncounted set.
+  /// This means that the result's `inserted` value is always `true`.
+  @inlinable
+  @discardableResult
+  public mutating func insert(_ newMember: __owned Element)
+  -> (inserted: Bool, memberAfterInsert: Element) {
+    _storage[newMember, default: 0] += 1
+    return (inserted: true, memberAfterInsert: newMember)
+  }
+
+  /// Removes the given element.
+  ///
+  /// If an element equal to `member` is contained in the set, its
+  /// multiplicity is decremented by one.
+  /// - Parameter member: An element to remove from the set.
+  /// - Complexity: O(*k*), where *k* is the number of unique elements in the
+  /// set, if the multiplicity of the given element was one. Otherwise, O(1).
```

Comment:
Nit: Hm, this complexity guarantee doesn't look quite right to me -- ignoring non-unique storage, removing a single item from a dictionary is expected to take O(1) hashing/equality checks on average, as long as `Element` properly implements hashing. 

It looks like the O(*k*) part comes from the `Dictionary.removeValue(forKey:)` docs, which seem to be bogus -- they aren't consistent with how we document the complexity of other dictionary operations.

(Strictly speaking, `removeValue(forKey:)` is indeed an O(n) operation, but so is `Dictionary.subscript(key:).getter`, `Set.contains(_:)` etc. -- hash tables have terrible worst-case performance. The same reasoning that allows us to say that updating a dictionary value is _expected to be_ an "O(1)" operation also applies to removals. There is a bit of handwaving involved.)

I recommend changing this (and every other primitive operation, from `contains(_:)` to `insert(_:)`) to say something like:

```swift
/// - Complexity: This operation is expected to perform O(1) hashing/equality checks on
///    average, provided that `Element` implements high-quality hashing.
```

(Note the careful wording -- `Equatable` doesn't guarantee that `==` takes O(1) time, so we cannot promise anything about how long dictionary operations will take in real time -- we can only put an upper bound on how many times they are expected to compare items. And of course random hash collisions make the worst case performance linear, so we need to talk about "expected" number of operations "on average". 😛)

Reply:
I spent a while pondering the time complexity here, and ultimately decided it was better to underpromise than overpromise. You wouldn’t want to do this in the middle of a loop, for instance.

Reply:
It's perfectly fine to remove items in a Dictionary in a loop! Removals are slower than insertions, but only by a (small) constant factor -- the two operations have exactly the same algorithmic complexity.

Reply:
Well, if `Dictionary.removeValue(forKey:) ` has a documented time complexity of O(_n_), I’m going to assume a future non-breaking update can actually make it O(_n_).